File: Freestylecoding.Math.FSharp.Tests/Natural/MultiplicativeIdentity.fs
namespace Natural

open Xunit
open Freestylecoding.Math

type public MultiplicativeIdentity() =
    static member GetMultiplicativeIdentity( this:'T when 'T :> System.Numerics.IMultiplicativeIdentity<'T,'T> ) : 'T =
        'T.MultiplicativeIdentity

    [<Fact>]
    member public this.IsIdentityCorrect () =
        Assert.Equal(
            Natural.Unit,
            MultiplicativeIdentity.GetMultiplicativeIdentity( Natural.Unit )
        )

File: Freestylecoding.Math.CSharp.Tests/Natural/MultiplicativeIdentity.cs
using Xunit;

namespace Natural.Tests;

using static Freestylecoding.Math.CSharp.Tests.Helpers;
using Natural = Freestylecoding.Math.Natural;

public class MultiplicativeIdentity {
	[Fact]
	public void IsIdentityCorrect() =>
		Assert.Equal(
			Natural.Unit,
			Natural.MultiplicativeIdentity
		);
}